test/functional/wallet_migration.py

```diff
@@ -0,0 +1,407 @@
+
+        addr_info = basic0.getaddressinfo(basic0.getnewaddress("", "bech32"))
+        assert_equal(addr_info["hdkeypath"], "m/84'/1'/0'/0/0")
+
+        self.log.info("Test migration of a basic keys only wallet with a balance")
+        basic1 = self.create_wallet("basic1")
+
+        for _ in range(0, 10):
+            default.sendtoaddress(basic1.getnewaddress(), 1)
+
+        self.generate(self.nodes[0], 1)
+
+        for _ in range(0, 5):
+            basic1.sendtoaddress(default.getnewaddress(), 0.5)
+
+        self.generate(self.nodes[0], 1)
+        bal = basic1.getbalance()
+        txs = basic1.listtransactions()
+
+        basic1.migratewallet()
+        assert_equal(basic1.getwalletinfo()["descriptors"], True)
+        self.assert_is_sqlite("basic1")
+        assert_equal(basic1.getbalance(), bal)
+        self.assert_list_txs_equal(basic1.listtransactions(), txs)
+
+        # restart node and verify that everything is still there
+        self.restart_node(0)
+        default = self.nodes[0].get_wallet_rpc(self.default_wallet_name)
+        self.nodes[0].loadwallet("basic1")
+        basic1 = self.nodes[0].get_wallet_rpc("basic1")
+        assert_equal(basic1.getwalletinfo()["descriptors"], True)
+        self.assert_is_sqlite("basic1")
+        assert_equal(basic1.getbalance(), bal)
+        self.assert_list_txs_equal(basic1.listtransactions(), txs)
+
+        self.log.info("Test migration of a wallet with balance received on the seed")
+        basic2 = self.create_wallet("basic2")
+        basic2_seed = get_generate_key()
+        basic2.sethdseed(True, basic2_seed.privkey)
+        assert_equal(basic2.getbalance(), 0)
+
+        # Receive coins on different output types for the same seed
+        basic2_balance = 0
+        for addr in [basic2_seed.p2pkh_addr, basic2_seed.p2wpkh_addr, basic2_seed.p2sh_p2wpkh_addr]:
+            send_value = random.randint(1, 4)
+            default.sendtoaddress(addr, send_value)
+            basic2_balance += send_value
+            self.generate(self.nodes[0], 1)
+            assert_equal(basic2.getbalance(), basic2_balance)
+        basic2_txs = basic2.listtransactions()
+
+        # Now migrate and test that we still see have the same balance/transactions
+        basic2.migratewallet()
+        assert_equal(basic2.getwalletinfo()["descriptors"], True)
+        self.assert_is_sqlite("basic2")
+        assert_equal(basic2.getbalance(), basic2_balance)
+        self.assert_list_txs_equal(basic2.listtransactions(), basic2_txs)
+
+    def test_multisig(self):
+        default = self.nodes[0].get_wallet_rpc(self.default_wallet_name)
+
+        # Contrived case where all the multisig keys are in a single wallet
+        self.log.info("Test migration of a wallet with all keys for a multisig")
+        self.nodes[0].createwallet(wallet_name="multisig0")
+        multisig0 = self.nodes[0].get_wallet_rpc("multisig0")
+        assert_equal(multisig0.getwalletinfo()["descriptors"], False)
```

Comment:
nit: (or `create_legacy_wallet`, as suggested above).

```suggestion
        multisig0 = self.create_wallet("multisig0")
```

Reply:
Done